samples/chat-app/server/DomainModel.fs

```diff
@@ -0,0 +1,106 @@
+namespace FSharp.Data.GraphQL.Samples.ChatApp
+
+open System
+
+//
+// Common model
+//
+type OrganizationId = OrganizationId of Guid
+type MemberId = MemberId of Guid
+type MemberPrivateId = MemberPrivateId of Guid
+type ChatRoomId = ChatRoomId of Guid
+type MessageId = MessageId of Guid
+
+type MemberRoleInChat =
+  | ChatAdmin
+  | ChatGuest
+
+type ChatRoomMessage =
+  { Id : MessageId
+    ChatRoomId : ChatRoomId
+    Date : DateTime
+    AuthorId : MemberId
+    Text : string }
+
+type ChatRoomSpecificEvent =
+  | NewMessage of ChatRoomMessage
+  | EditedMessage of ChatRoomMessage
+  | DeletedMessage of MessageId
+  | MemberJoined of MemberId * string
+  | MemberLeft of MemberId * string
+
+type ChatRoomEvent =
+  { ChatRoomId : ChatRoomId
+    Time : DateTime
+    SpecificData : ChatRoomSpecificEvent }
+
+//
+// Persistence model
+//
+type Member_In_Db =
+  { PrivId : MemberPrivateId
+    Id : MemberId
+    Name : string }
+
+type ChatMember_In_Db =
```

Comment:
I would suggest not using snake case here to keep consistency with naming conventions.

Reply:
This would be fine I suppose
```suggestion
type ```ChatMember In Db`` =
```

Reply:
@ivelten I get your point, but the reason why I think it makes sense to use this other notation here is precisely to make it clear that it's not a usual name. The prominent `_In_Db` suffix is supposed to serve as metadata: this type is not to be mistaken for a regular domain model type, but an internal type you use to carry information in the same shape available in the database.
If the type were named `MemberInDb` this extra semantic layer would not be as striking to the eye as having a rather awkward `_In_Db` suffix. 
By that, I don't mean we should start using these prefixes for every type we define just to advertise any extra information we may find valuable.
In contrast, see how I named the types `MeAsAMember` and `MeAsAChatMember` (instead of, say, `Me_As_A_Chat_Member`). Although the `AsAChatMember` part may be seen as a suplement, it is really a complement to the type's name and not technical metadata.

@xperiandri I like the three backticks notation for functions, specially (unit) test functions, but I have to admit that I find it kind of cumbersome (and even cryptic) for types. Since the name is not that long anyway, I'd vote for the more simplistic snake case approach.

Reply:
Your idea looks logical